src/commands/talk.ts

```diff
@@ -0,0 +1,224 @@
+		if (!permissions) {
+			throw new Error('Could not fetch permissions for channel');
+		}
+		if (!permissions.has(PermissionFlagsBits.Connect)) {
+			throw new Error('Missing required permissions to connect to voice channel');
+		}
+		if (!permissions.has(PermissionFlagsBits.Speak)) {
+			throw new Error('Missing required permissions to speak in voice channel');
+		}
+
+		// check to see if we're already connected to this channel
+		// a channel can only have one connection & one player at a time!
+		if (getVoiceConnection(channel.guild.id)) {
+			throw new Error('Already talking in channel\n\nPlease try again later');
+		}
+
+		// store the audio buffer in a temporary .wav file to pass to discordjs/voice
+		const tempFile = fileSync({ prefix: 'dectalk', postfix: '.wav' });
+		writeFileSync(tempFile.name, wavData);
+		const stream = createReadStream(tempFile.name);
+
+		// create an audio resource from the temporary file
+		let resource: AudioResource;
+		try {
+			resource = createAudioResource(stream);
+		} catch (error) {
+			const error_ = error as Error;
+			// For this error, provide a more useful message
+			if (error_.message === 'FFmpeg/avconv not found!') {
+				throw new Error(
+					"'ffmpeg-static' missing proper encoder for current OS\n\n" +
+						'Please run a clean install of all dependencies'
+				);
+			}
+
+			// remember to close file stream
+			stream.close();
+
+			// Don't tamper with any other errors
+			throw error;
+		}
+
+		// create a player to stream the resource
+		const player = createAudioPlayer({
+			behaviors: {
+				// play even when there's no one in the channel
+				noSubscriber: NoSubscriberBehavior.Play,
+			},
+		});
+
+		// create a new audio connection to the voice channel the command was given in
+		log('joining channel');
+		const connection = joinVoiceChannel({
+			channelId: channel.id,
+			guildId: channel.guild.id,
+			adapterCreator: channel.guild.voiceAdapterCreator,
+			selfMute: false,
+		});
+
+		// when the connection is ready, attach the player and give the player the audio
+		connection.once(VoiceConnectionStatus.Ready, () => {
+			log('connection ready');
+			connection.subscribe(player);
+			player.play(resource);
+		});
+
+		// when the player finishes playing the audio, disconnect if connected, and delete the audio file
+		player.on(AudioPlayerStatus.Idle, () => {
+			log('player done');
+
+			if (
+				connection.state.status !== VoiceConnectionStatus.Disconnected &&
+				connection.state.status !== VoiceConnectionStatus.Destroyed
+			) {
+				connection.disconnect();
+			}
+
+			// remember to close file stream
+			stream.close();
+		});
+
+		// when the connection closes, stop the player if playing and end the interaction
+		connection.on(VoiceConnectionStatus.Disconnected, () => {
+			log('disconnected');
+			connection.destroy();
+
+			log('replying to interaction');
+			void reply({
+				content: message,
+			});
+
+			player.stop();
```

Comment:
Does this fire the `AudioPlayerStatus.Idle` event on the player, or should we also close the stream here?

Reply:
Yes, stopping the player fires the Idle event.